src/esp/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# https://embeddedartistry.com/blog/2017/3/7/clang-weverything
+# lots of warnings and all warnings as errors
+# TODO(msb) turn on -Wextra
+add_compile_options(-Wall -Werror)
```

Comment:
Please don't :) `-Werror` breaks as soon as a new version of a compiler decides to throw warnings for something that didn't warn before. Or older compilers tend to warn on something that is treated as a standard practice in later versions and is just too annoying in the long run. We should still strive for a zero-warning build, but without suffering too much :)

This is also Clang- ang GCC-specific and since we'll have a MSVC build soon (#84 and mosra/magnum-bindings#2), let's not make the porting harder for no reason ;) What I suggested elsewhere and have in TODOs in #42 is `CORRADE_USE_PEDANTIC_FLAGS` -- it contains a broad set of useful warnings, works on MSVC as well (and also disables ones that are just noise). It can be used like this:

```cmake
set_directory_properties(PROPERTIES CORRADE_USE_PEDANTIC_FLAGS ON)
```

Oh -- this got already used on a few places since #91, so if you can, search for `CORRADE_USE_PEDANTIC_FLAGS`
and remove the other occurences, as they would be redundant now :) Also it enables `-Wextra`, so it might bring you a bit more work.

Reply:
Sure, just saw the commit -- why not go all the way and enable `CORRADE_USE_PEDANTIC_FLAGS`, then? Did it result it many more complaints? :)

Reply:
OK, makes sense :+1: I was under a false assumption that it got in a pretty good shape already :)